kuka_sunrise_control/robot_control/src/robot_control/rate_scaled_controller.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright 2020 Zoltán Rési
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "robot_control/rate_scaled_controller.hpp"
+
+#include <sys/mman.h>
+#include <string>
+#include <memory>
+#include <vector>
+
+namespace robot_control
+{
+
+ScaledJointController::ScaledJointController(
+  const std::string & node_name,
+  const rclcpp::NodeOptions & options)
+: InterpolatingController(node_name, options)
+{
+  auto set_rate_callback = [this](
+    kuka_sunrise_interfaces::srv::SetDouble::Request::SharedPtr request,
+    kuka_sunrise_interfaces::srv::SetDouble::Response::SharedPtr response) {
+      int cmd_per_frame = static_cast<int>(JointControllerBase::ms_in_sec_ /
+        loopPeriod() / (8 * request->data)) + 1;
```

Comment:
What is the magic number 8? Publish frequency of camera? If yes, this hidden dependency should be generalized out somehow.